[Tan Tze Yeong] iP

tantzeyeong opened the pull request:
# Duke

Duke frees your mind and lets you focus on the actual tasks! It's,

- text-based
- simple
- ~~fast~~ too fast! Works at *LIGHTSPEED* ⚡

All you got to do is,

1. download it from [here](https://github.com/tantzeyeong/ip/releases).
2. double-click it.
3. add your tasks.
4. let Duke remember them for you! 😄 

And guess what? It's completely **FREE**!! 😱 

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you're a Java programmer, you can practice Java with it too! Here's the `main` method:

```java
public static void main(String[] args) {
    new Duke("data/Duke.txt").run();
}
```

tantzeyeong committed:
add greetings to user

tantzeyeong committed:
add echoing of user commands

tantzeyeong committed:
add "add" and "list" user commands

tantzeyeong committed:
add Task class, add "markAsDone" and "markAsNotDone" of tasks

tantzeyeong committed:
add ToDos, Deadlines, Events

tantzeyeong committed:
add error handling

tantzeyeong committed:
add delete function

tantzeyeong committed:
Refactor printing code

tantzeyeong committed:
Add automated text UI testing

tantzeyeong committed:
Rename bot name

tantzeyeong committed:
Add save feature, extract classes, add Command class, organize classes into packages

tantzeyeong committed:
Add date recognition for deadline and event tasks

tantzeyeong committed:
Merge branch 'branch-Level-7'

tantzeyeong committed:
Merge branch 'branch-Level-8'

tantzeyeong committed:
Remove commented out code

tantzeyeong committed:
Fix loading of deadlines and events due to change in date format

tantzeyeong committed:
Add gradle support

tantzeyeong committed:
Add gradle support

tantzeyeong committed:
Change mainClassName in build.gradle

tantzeyeong committed:
Add JUnit tests

tantzeyeong committed:
Track MANIFEST.MF

tantzeyeong committed:
Add finding of task by keyword

tantzeyeong committed:
Add java docs

tantzeyeong committed:
Comply with coding standard

tantzeyeong committed:
Merge branch 'branch-A-CodingStandard'

tantzeyeong committed:
Merge branch 'branch-Level-9'

tantzeyeong committed:
Merge branch 'branch-A-JavaDoc'

tantzeyeong committed:
Update java docs

tantzeyeong committed:
Comply with coding standards

tantzeyeong committed:
Add checkstyle into project

tantzeyeong committed:
Update build.gradle for setting up Java FX

dlimyy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,54 @@
+import duke.command.Command;
+import duke.model.*;
```
Wildcard(aka *) should be avoided here so the classes should be stated explicitly.

tantzeyeong replied:
Oh thank you for pointing that out!

dlimyy commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,37 @@
+ * The command to be executed with respect to the user's input.
+ */
+public abstract class Command {
+    private Boolean isExit = false;
```
Good naming for Boolean variables!

dlimyy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,94 @@
+        String[] description;
+
+        switch (keyword) {
+            case BYE:
```
There should not be any indentation for the cases clauses, so the word 'case' should have the same indentation as switch

dlimyy commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,94 @@
+
+        String[] description;
+
+        switch (keyword) {
```
Good use of switch cases to handle different commands!

dlimyy reviewed (commented):
Overall, the code quality is good and good coding practices are well adhered to!

tantzeyeong committed:
Add GUI to project

tantzeyeong committed:
Merge branch 'branch-Level-10'

tantzeyeong committed:
Update mainClassName in build.gradle

The mainClassName in build.gradle was left as "Duke" since the last time
a JAR file was created, which is incorrect.

The newly created JAR file would be unable to run as main cannot be
found in "Duke".

Let's update the mainClassName from "Duke" to "Launcher".

tantzeyeong committed:
Enable assertions

tantzeyeong committed:
Add assertions to Commands

tantzeyeong committed:
Refactor code quality

tantzeyeong committed:
Update assertion message

tantzeyeong committed:
Merge pull request #1 from tantzeyeong/branch-A-Assertions

Add assertions to project

tantzeyeong committed:
Merge branch 'master' into branch-A-CodeQuality

tantzeyeong committed:
Merge pull request #2 from tantzeyeong/branch-A-CodeQuality

Refactor code quality

tantzeyeong committed:
Add more exceptions in Parser class

The chatbox still does not handle some errors caused by incorrect user
inputs.

The chatbox crashes instantly when these errors are not handled.

Adding more try-catch blocks and throwing more exceptions when invalid
inputs are given in the Parser class can stop the program from crashing.

Let's throw more exceptions in the Parser class and catch them in the
Duke class.

tantzeyeong committed:
Rename variables in TaskList

tantzeyeong committed:
Add Continuous Integration into project

tantzeyeong committed:
Add UndoCommand to chatbot

The user is unable to conveniently undo his mistake after inputting a
command that changes the state of the tasklist.

In cases where the tasklist starts to get very long, the user may have a
hard time looking for the last change he did to manually revert the
change himself.

Let's add a new UndoCommand to allow the user to revert the tasklist to
the state before the previous change occurred.

tantzeyeong committed:
Change permission of gradlew

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Disable checkstyles

tantzeyeong committed:
Disable checkstyles

tantzeyeong committed:
Enable checkstyles

tantzeyeong committed:
Merge pull request #4 from tantzeyeong/branch-C-Undo

Add UndoCommand to chatbot

tantzeyeong committed:
Add more error handling in Parser class

tantzeyeong committed:
Refactor code for error handling in Parser class

tantzeyeong committed:
Add alternate response for when FindCommand fails to match any tasks

tantzeyeong committed:
Add more JUnit tests

tantzeyeong committed:
Add personality to Duke

tantzeyeong committed:
Fix ExitCommand to automatically exit program

tantzeyeong committed:
Add more assertions

tantzeyeong committed:
Improve GUI of Duke

tantzeyeong committed:
Add more error handling

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Fix IO redirection test error

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Add user guide

tantzeyeong committed:
Add Ui.png to docs folder

tantzeyeong committed:
Update user guide

tantzeyeong committed:
Fix file path for duke.txt

tantzeyeong committed:
Fix checkstyle errors

tantzeyeong committed:
Update testing

tantzeyeong committed:
Add acknowledgements
